frontend/lib/widgets/app_bars.dart

```diff
@@ -84,12 +84,20 @@ class SearchSliverAppBarState extends State<SearchSliverAppBar> {
   final TextEditingController textEditingController = TextEditingController();
   final FocusNode focusNode = FocusNode();
 
+  var keyboardToggler = false;
+
   @override
   Widget build(BuildContext context) {
     final t = context.t;
     final foregroundColor = Theme.of(context).appBarTheme.foregroundColor;
     return SliverAppBar(
       title: TextField(
+        onTap: () {
+          keyboardToggler = true;
+        },
+        onTapOutside: (PointerDownEvent event) {
```

Comment:
if i do only
```js
        onTapOutside: (PointerDownEvent event) {
          FocusScope.of(context).unfocus();
        },
```
and revert all other changes it also seem to work as expected. Have i overseen anything that makes these flags required?

Reply:
Yes, if you do so, the search button won't toggle (open-hide-open-hide) the keyboard anymore.

Current behavior (before my changes) is:
If you click search button once - it opens the keyboard.
If you click it again - it hides the keyboard.

I assumed that was the intent, and added a flag to preserve this behavior.

Reply:
on ios the open and closing by clicking on the icon doesn't work. As we discussed the icon is not really needed because the search happens on change of the text input, so we may change the button to an icon

Reply:
What do you think if we keep it as button, but it won't do anything, but splash?
Like, for those, who like to click on buttons after typing to search, even when the automatic search is working, just to make sure?

_Use case_:
You entered some value in the search field and didn't notice the search results updated.
Then you click on our magic "search" button, it splashes, and that gives you confidence that everything is okay, the search has been applied.

Reply:
i think i wouldn't just pass an empty function for the splash animation. I think its dirty to pattern :D
The icon in this case is even at the moment just an indicator for the search
I think we should basically use an icon with no function
```js
Padding(
          padding: EdgeInsets.only(right: 16),
          child: Icon(Icons.search),
        ),
```

Reply:
Pushed the icon with some opacity, to make it different from the real "clear" button

Reply:
One more idea would maybe be to add the search icon in the `icon` property of the InputDecoration of the TextField. Maybe that would move the search icon to the left, but maybe that's actually better anyway? At least in many google apps the icon is to the left of the input field (if there is any).

Reply:
Or maybe even replace the whole input with a flutter [SearchBar](https://api.flutter.dev/flutter/material/SearchBar-class.html) (which didn't exist back in the day this page was first developed - not sure if it fits here).

Reply:
i think a search bar could be discussed and implemented in a separate task if we want it.